lib/cocoapods/installer.rb

```diff
@@ -377,7 +377,13 @@ def create_pod_installer(pod_name)
         end
       end
 
-      raise Informative, "Could not install '#{pod_name}' pod. There is no target that supports it." if specs_by_platform.empty?
+      if specs_by_platform.empty?
+        requiring_targets = pod_targets.select { |pt| pt.recursive_dependent_targets.any? { |dt| dt.pod_name == pod_name } }
+        message = "Could not install '#{pod_name}' pod"
+        message += ", dependended upon by #{requiring_targets.to_sentence}" unless requiring_targets.empty?
+        message += ". There is either no platform to build for, or no target to build."
+        raise StandardError, message
```

Comment:
not `Informative`?

Reply:
No -- this is not an expected failure mode, and should _never_ be hit, and we should get an issue when it is hit, so we can surface the underlying issue

Reply:
sounds good